src/lib389/lib389/cli_ctl/dblib.py

```diff
@@ -48,6 +62,47 @@ def __init__(self, *args, **kwargs):
         self.__dict__ = self
 
 
+def get_bdb_impl_status():
+    # Check installed rpms
+    try:
+        from rpm import TransactionSet
+        ts = TransactionSet()
+        pkgs = { pkgname:(len ([ pkg for pkg in ts.dbMatch( 'name', \
+                 pkgname ) ]) > 0) for pkgname in TESTED_PACKAGES}
+        ts.closeDB()
+    except ImportError:
+        if os.path.isfile(RPM_TOOL):
+            pkgs = { pkgname:( subprocess.run([RPM_TOOL, '-q', pkgname], \
+                     stdin=subprocess.DEVNULL,stderr=subprocess.DEVNULL \
+                     ).returncode  == 0 ) for pkgname in TESTED_PACKAGES }
+        else:
+            pkgs = { pkg:None for pkg in TESTED_PACKAGES}
+
+    # Check libback-ldbm plugin content
+    p = Paths()
+    plugin = None
+    BDB_IMPL_STATUS
+    for pdir in PLUGIN_PATHS:
+        for ppath in glob.glob(f'{pdir}/libback-ldbm.*'):
+            plugin = ppath
+    if plugin and os.path.isfile(OBJDUMP_TOOL):
+        rc = subprocess.run(['objdump', '-R', plugin], text=True, capture_output=True)
+        symbs = { symb: symb in rc.stdout for symb in TESTED_SYMBOLS }
+    else:
+        symbs = { symb: None for symb in TESTED_SYMBOLS }
```

Comment:
I think it'll be nice (but not required, - it's up to you) to make these two code blocks a separate helper function for readability and better maintainability.

LIke:
```
    pkgs = check_installed_packages()
    symbs = check_plugin_symbols()
```

Reply:
I will do that but probably better to use:
```
    pkgs = check_installed_packages(TESTED_PACKAGES)
    symbs = check_plugin_symbols('libback-ldbm', TESTED_SYMBOLS)
```
To have reusable code (and move the helper function in utils.py